compiler/src/org.graalvm.compiler.replacements/src/org/graalvm/compiler/replacements/SnippetTemplate.java

```diff
@@ -558,13 +558,15 @@ public ValueNode findLength(FindLengthMode mode, ConstantReflectionProvider cons
         private final GuardsStage guardsStage;
         private final LoweringTool.LoweringStage loweringStage;
         private int hash;
+        private boolean expandsVolatileMembarsLate;
```

Comment:
This `expandsVolatileMembarsLate` field must participate in `CacheKey.equals` and `CacheKey.hashCode`.

Reply:
Right. loweringState is not hashed with the other fields. I suppose it's ok to leave expandsVolatileMembarsLate and only change equals()?

Reply:
Yes.